blocks/autocompleters/index.js

```diff
@@ -107,8 +107,8 @@ export function blockAutocompleter( { onReplace } ) {
  * @returns {Completer} Completer object used by the Autocomplete component.
  */
 export function userAutocompleter() {
-	const getOptions = () => {
-		return ( new wp.api.collections.Users() ).fetch().then( ( users ) => {
+	const getOptions = ( search ) => {
+		return ( new wp.api.collections.Users() ).fetch( search ? { data: { search } } : null ).then( ( users ) => {
```

Comment:
This is a long line. Splitting out the ternary into variable assignment could improve this.

```js
let payload;
if ( search ) {
	payload = { data: { search } };
}

return ( new wp.api.collections.Users() ).fetch( payload ).then( ( users ) => {
```

Reply:
good idea, implemented in https://github.com/WordPress/gutenberg/pull/4406/commits/ac62c0098e0a40ace0ec656b37958ea1fec3727d